documentation/src/docs/security/generating-tls-artifacts.md

```diff
@@ -0,0 +1,324 @@
+   * An X.509 certificate containing the public key identifying the server
+   * A private key for the public/private key pair
+
+   ```bash
+   $ keytool -storetype JKS -keystore <java-keystore-file-name> -storepass <java-keystore-password>\
+           -genkey -keyalg <key-algorithm> -keysize <key-size> -keypass <private-key-password>\
+           -alias <alias> -validity <validity>\
+           -dname "<distinguished-name>"\
+           -ext SAN=dns:<hostname1>,dns:<hostname2>,ip:<ipaddress1>
+   ```
+   
+   Example:
+   ```bash
+   $ keytool -storetype JKS -keystore server_unsigned.keystore.jks -storepass changeit\
+              -genkey -keyalg RSA -keysize 2048 -keypass changeit\
+              -alias server -validity 365\
+              -dname "CN=server.pravega.io, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown"\
+              -ext SAN=dns:server.abc.com,ip:127.0.0.1
+   
+   # Optionally, verify the contents of the generated file
+   $ keytool -list -v -storetype JKS -keystore server_unsigned.keystore.jks -storepass changeit
+   ```
+
+2. Generate a Certificate Signing Request (CSR) for the server certificate.
+
+   It helps to think of a CSR as an application for getting a certificate signed by a trusted authority. 
+   
+   A CSR is typically generated on the same server/node that uses the certificate, so that the corresponding private key
+   doesn't need to be moved anywhere else. In some other environments, CSRs are generated in a central server and the 
+   resulting certificates are distributed to the services that need them.  
+
+   ```bash
+   $ keytool -keystore <java-keystore-file-path> -storepass <java-keystore-password> -alias <alias> \
+             -certreq -file <csr-file-path> \
+             -storepass <java-keystore-password>
+   ```
+   
+   Example:
+   ```bash
+   $ keytool -keystore server_unsigned.keystore.jks -storepass changeit -alias server \ 
+             -certreq -file server.csr \
+             -ext SAN=dns:server.abc.com,ip:127.0.0.1
+   ```
+
+3. Submit the CSR to a CA and obtain a signed certificate.
+
+   If you are using a public or internal CA service, follow that CA's process for submitting the CSR and obtaining
+   a signed certificate. To use the custom CA generated using the steps mentioned [earlier](#stage-1-setting-up-a-certificate-authority-ca) or an internal CA
+   certificate/key bundle, use the steps below:
+   
+   First create a `server-csr.conf` file with the following contents. Replace `alt_names` with the server's hostname/ 
+   IP address that Pravega will use to access the server. If the same certificate is used in multiple servers, add 
+   hostnames/IP addresses of all the servers. 
+   
+   ```
+   [req]
+   req_extensions = v3_req
+   prompt = no
+   
+   [v3_req]
+   subjectAltName = @alt_names
+   
+   [alt_names]
+   DNS.1 = server.abc.com
+   DNS.2 = server.pravega.io
+   IP.1 = 127.0.0.1
+   ```
+   
+   Now, have the CA to sign the certificate:
```

Comment:
"have the CA to sign" -> "have the CA sign"

Reply:
Fixed.